### PR TITLE
Use surface extents for swap chain extents.

### LINKED ISSUE #98779: Failure to create swap chain since swappy additions (Linux Wayland).
### Tested versions

Introduced in:
748f4079e387fb555ee64c1619e94ba099bfa77a 

Reproduced on current master:
1bffd6c73b44b85e5889f54e14b2193940cf5bb1 (m

### System information

Godot v4.4.dev (1bffd6c73) - Fedora Linux 41 (Workstation Edition) on Wayland - Wayland display driver, Single-window, 2 monitors - Vulkan (Forward+) - dedicated AMD Radeon RX 6700 XT (RADV NAVI22) - 13th Gen Intel(R) Core(TM) i7-13700K (24 threads)

### Issue description

```
Godot Engine v4.4.dev.custom_build.1bffd6c73 (2024-11-02 20:59:31 UTC) - https://godotengine.org
WARNING: Can't obtain the XDG decoration manager. Libdecor will be used for drawing CSDs, if available.
     at: init (platform/linuxbsd/wayland/wayland_thread.cpp:3731)
libdecor-gtk-WARNING: Failed to initialize GTK
Failed to load plugin 'libdecor-gtk.so': failed to init
Vulkan 1.3.289 - Forward+ - Using Device #0: AMD - AMD Radeon RX 6700 XT (RADV NAVI22)

ERROR: Condition "err != VK_SUCCESS" is true. Returning: ERR_CANT_CREATE
   at: swap_chain_resize (drivers/vulkan/rendering_device_driver_vulkan.cpp:3003)
ERROR: Condition "err != VK_SUCCESS" is true. Returning: ERR_CANT_CREATE
   at: swap_chain_resize (drivers/vulkan/rendering_device_driver_vulkan.cpp:3003)
ERROR: Condition "err != VK_SUCCESS" is true. Returning: ERR_CANT_CREATE
   at: swap_chain_resize (drivers/vulkan/rendering_device_driver_vulkan.cpp:3003)
```

It appears the wrong extent is used which in this case has invalid sizes.  Will open PR.

### Steps to reproduce

Just create a new project using Forward+ renderer.

### Minimal reproduction project (MRP)

N/A

### PR DESCRIPTION
**Pull request by tdaven**
Fixes https://github.com/godotengine/godot/issues/98779

This PR just uses the calculated surface extents instead of the current extents which are sometimes the special value UINT32_MAX.

### CODE REVIEW

File: drivers/vulkan/rendering_device_driver_vulkan.cpp
Comment (tdaven) [inline drivers/vulkan/rendering_device_driver_vulkan.cpp:2992]: Line 2901 would handle the case of UINT32_MAX extents from GetPhysicalDeviceSurfaceCapabilitiesKHR that means "indicating that the surface size will be determined by the extent of a swapchain targeting the surface".  This would be stored to native_display_size but then not updated when the others are handled.  

Since native_display_size wasn't used anywhere else it was simple to just use the calculated extents instead.

**Review by akien-mga — APPROVED** [review]
I confirm this fixes the linked issue for me.

Still needs an OK from the @godotengine/rendering team.
CC @darksylinc

**Comment by darksylinc** [comment]
Hold off the merge for a little bit. I need to check if it's correct.

The main issue is that on Android we will often get `swap( width, height )` (because the swapchain is rotated 90°) and eyeballing the patch it seems that will break. I need to look into this further.

**Comment by Alex2782** [comment]
Last week a forum user reported this error, Android Editor `swappy=yes`. But with `swappy=no` it stopped occurring.
_(I couldn't reproduce this error)_ at line 2886

image

**Comment by darksylinc** [comment]
Hi

> Last week a forum user reported this error, Android Editor swappy=yes. But with swappy=no it stopped occurring.

That problem is unrelated to this PR, please open a new ticket providing all the requested info so we can make a blocklist to disable swappy for that particular device(s).

**Comment by akien-mga** [comment]
Superseded by #98852. Thanks for the contribution!

**Pull request closed without merging**
